subsys/bluetooth/mesh/settings.c

```diff
@@ -186,8 +186,7 @@ static void store_pending(struct k_work *work)
 {
 	LOG_DBG("");
 
-	if (IS_ENABLED(CONFIG_BT_MESH_RPL_STORAGE_MODE_SETTINGS) &&
-	    atomic_test_and_clear_bit(pending_flags, BT_MESH_SETTINGS_RPL_PENDING)) {
+	if (atomic_test_and_clear_bit(pending_flags, BT_MESH_SETTINGS_RPL_PENDING)) {
```

Comment:
Hm... what will happen if I have the enabled settings subsystem and want to use the custom way to store RPL? for example in external flash.

Shouldn't it be a separate option for this choice: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/mesh/Kconfig#L577-L588 ?
I guess options (settings\custom\any others) should exclude each other.

If the custom option was chosen then the storage function shouldn't exist at all.

Reply:
For the custom RPL to use this branch you still need to call `bt_mesh_settings_store_schedule(BT_MESH_SETTINGS_RPL_PENDING);` which is only done internally (e.g. rpl.c calls this when needs to store updated entries). If you don't call `bt_mesh_settings_store_schedule(BT_MESH_SETTINGS_RPL_PENDING);`, this never gets triggered. So , it is up to implementation whether to use it or not. I don't see a need in extra Kconfig, just for this if-statement?

Reply:
This was born out of a discussion Pavel and I had of how a custom RPL implementation would synchronize with the storing of other settings entries for the mesh. My thinking was that assuming the operation was interrupted by a power loss the least harmful way to go is to have the RPL written back first during both an IV update and an deprovisioning of the device. As it stands now, there is no way to accomplish this (short of immediately writing back all RPL updates, which would block the mesh stack) without modifying the subsystem code. This at least allows a synchronization point if need be for asynchronous implementations like I was trying to write.

Considering that it's opt in to set the pending settings flag and we still have to define the API function in all the non-settings implementations of the RPL I don't think there is a concern other than maybe some unclear naming here (It's not immediately obvious to me that using the settings subsystem would be required for the pending store to be called).

Reply:
I'm pretty sure it should be supported correctly but it is Ok for now. 
Anycase it will be changed if we or somebody else decide to add external flash support.